### PR TITLE
Solving default frame consistency in config files

### PR DESCRIPTION
**Pull request by AlexisTM**
Solving default frame consistency in config files #789 

* Changing the `local_origin` frame to `map` for consistency.
* Correction of the `global_position/frame_id` and `local_position/frame_id` to map frame

### CODE REVIEW

**Comment by AlexisTM** [comment]
I updated the APM config as well. @TSC21

**Comment by TSC21** [comment]
Where you see `fcu_utm` please update to `fcu`

**Comment by AlexisTM** [comment]
Should check for the time reference. 

Is this related to tf?

```
time:
  time_ref_source: "fcu"  # time_reference source
```

**Comment by TSC21** [comment]
> Should check for the time reference. 

Keep it as `fcu`. That has nothing to do with frames of reference.

File: mavros/launch/px4_config.yaml [hunk 1]
```diff
@@ -44,18 +44,18 @@ cmd:
...
 global_position:
-  frame_id: "fcu"           # pose and fix frame_id
+  frame_id: "map"           # pose and fix frame_id
   rot_covariance: 99999.0   # covariance for attitude?
   use_relative_alt: true    # use relative altitude for local coordinates
   tf:
     send: false               # send TF?
-    frame_id: "local_origin"  # TF frame_id
+    frame_id: "map"  # TF frame_id
     global_frame_id: "earth"  # TF earth frame_id
```
Comment (TSC21) [inline mavros/launch/px4_config.yaml]: Can you add `global_frame_id` also to apm_config please?

**Comment by TSC21** [comment]
@AlexisTM looking good but can you please run a sanity check on the source code also and match the variables, that represent the parameters, to the config files? Basically the same you did above but for the cpp files. Thanks

**Comment by AlexisTM** [comment]
The only parameters not synced are in `sys_status.cpp`

https://github.com/mavlink/mavros/blob/d283fb0a3f094caa4282922351fb965f69f9c898/mavros/src/plugins/sys_status.cpp#L431-L433

* `global_position/tf/send` default to `false`
* `imu, checked
* `local_position/tf/send` default to `false`
* `local_position/tf/send_fcu` default to `false`
* `mission/pull_after_gcs` default to `true`

**Comment by TSC21** [comment]
> The only parameters not synced are in `sys_status.cpp`

Can you sync with the values that are on the `.cpp` file, please?

> Note that the builds are failing because of the MAVLink signing development.

Yeah we need that the release comes out on the deb pakages.

**Comment by AlexisTM** [comment]
* Added sys_status, and mavros_extras plugins.
* Added the FOV for the Px4flow (6.8 degrees at 5 meters)
* Adding the copter_visualization configuration in the yaml file

**Comment by TSC21** [comment]
@AlexisTM `copter_visualization` is a separate node that can be run in parallel. So the parameters are  not loaded from the config files of the `mavros` node. The parameter config should got: or to a `launch` file that launches the `copter_visualization` node, or to a `yaml` file on the launch folder and getting the parameters on the cpp file through `XmlRpc::XmlRpcValue` (check how it is done on `servo_state_publisher.cpp`).
The rest of the changes though seem consistent. Thanks

**Comment by TSC21** [comment]
Also, can you please change the `vibration` `frame_id` on both plugin and yaml to `base_link`? Thanks

**Comment by AlexisTM** [comment]
I think it is all fine with the way the parameters are read for the `copter_visualization`. Indeed, there are only a finite number of parameters, I would propose to avoid the use of `XmlRpc::XmlRpcValue`. 

Setting the parameters in a yaml file or the launch file is sufficient.

File: mavros/launch/apm_config.yaml [hunk 2]
```diff
@@ -167,7 +169,7 @@ fake_gps:
 # px4flow
 px4flow:
   frame_id: "px4flow"
-  ranger_fov: !degrees 0.0  # XXX TODO
+  ranger_fov: !degrees 6.8  # XXX TODO
```
Comment (TSC21) [inline mavros/launch/apm_config.yaml]: You can remove the `TODO` comment now

Comment (TSC21) [reply mavros/launch/apm_config.yaml]: `6.8 degreens at 5 meters, 31 degrees at 1 meter`

File: mavros_extras/launch/px4flow_config.yaml [hunk 3]
```diff
@@ -25,7 +25,7 @@ image:
 
 px4flow:
   frame_id: "px4flow"
-  ranger_fov: !degrees 0.0  # XXX TODO
+  ranger_fov: !degrees 6.8  # XXX TODO
```
Comment (TSC21) [inline mavros_extras/launch/px4flow_config.yaml]: `6.8 degreens at 5 meters, 31 degrees at 1 meter`

File: mavros_extras/src/plugins/px4flow.cpp [hunk 4]
```diff
@@ -45,7 +45,9 @@ class PX4FlowPlugin : public plugin::PluginBase {
 		flow_nh.param<std::string>("frame_id", frame_id, "px4flow");
 
 		/** Default rangefinder is Maxbotix HRLV-EZ4 */
-		flow_nh.param("ranger_fov", ranger_fov, 0.0);	/** @todo Check MAxbotix HRLV-EZ4 Field-of-View */
+		flow_nh.param("ranger_fov", ranger_fov, 0.119428926);	/** @todo Check MAxbotix HRLV-EZ4 Field-of-View
```
Comment (TSC21) [inline mavros_extras/src/plugins/px4flow.cpp]: Replace with
```c++
/**
 * @note This is a narrow beam (60cm wide at 5 meters,
 * but also at 1 meter) 6.8 degrees at 5 meters, 31 degrees at 1 meter
 */
flow_nh.param("ranger_fov", ranger_fov, 0.119428926);
```

**Comment by AlexisTM** [comment]
You are welcome. ;)

**Pull request merged**